### ISSUE TITLE
[PRE REVIEW]: JavaPermutationTools: A Java Library of Permutation Distance Metrics

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#887]
**Submitting author:** @cicirello (<a href="http://orcid.org/0000-0003-1072-8559">Vincent A Cicirello</a>)
**Repository:** <a href="https://github.com/cicirello/JavaPermutationTools" target ="_blank">https://github.com/cicirello/JavaPermutationTools</a>
**Version:** 1.0
**Editor:** Pending
**Reviewer:** Pending

**Author instructions**

Thanks for submitting your paper to JOSS @cicirello. The JOSS editor (shown at the top of this issue) will work with you on this issue to find a reviewer for your submission before creating the main review issue.

@cicirello if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00887/joss.00887/10.21105.joss.00887.pdf)

**Comment by cicirello** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

🚧 🚧 🚧 Experimental Whedon features 🚧 🚧 🚧

# Compile the paper
@whedon generate pdf

```

**Comment by cicirello** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00887/joss.00887/10.21105.joss.00887.pdf)

**Comment by cicirello** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00887/joss.00887/10.21105.joss.00887.pdf)

**Comment by cicirello** [comment]
@whedon list editors

**Comment by whedon** [comment]
Current editors:

```
@Kevin-Mattheus-Moerman
@acabunoc
@arfon
@arokem
@brainstorm
@cMadan
@danielskatz
@george-githinji
@jakevdp
@jasonclark
@jedbrown
@karthik
@katyhuff
@kyleniemeyer
@labarba
@leeper
@lheagy
@mgymrek
@pjotrp
@tracykteal
```

**Comment by yochannah** [comment]
@whedon assign @yochannah as editor

**Comment by whedon** [comment]
OK, the editor is @yochannah

**Comment by yochannah** [comment]
@cicirello thanks for submitting your paper! Let me know if you have any suggestions for possible reviewers.

**Comment by yochannah** [comment]
I've also contacted @Nikoleta-v3 to see if she's interested in reviewing.

**Comment by cicirello** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00887/joss.00887/10.21105.joss.00887.pdf)

**Comment by cicirello** [comment]
@yochannah  Thanks.  I'll take a look at the list of reviewers. Also, since submitting I actually added a package of distance metrics on sequences to the library, and just added a short paragraph toward the end of the paper to indicate such.  Anyway, the version number when I submitted JOSS paper was 1.0, but current version number is now 1.1 (in case that matters for the review).

**Comment by cicirello** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by yochannah** [comment]
@alex-konovalov do you know anyone who might be able to review a java package?

**Comment by yochannah** [comment]
👋  Hey @hausen, might you be interested in reviewing this?

**Comment by ChrisJefferson** [comment]
I am happy to review this package. I know Java quite well, but am a little rusty with building / classpaths / junit.

@cicirello  Could you please add explicit instructions for building the package, running the tests and running the examples?

**Comment by cicirello** [comment]
@ChrisJefferson Yes, I'll add instructions for building, running the tests, running the examples.  I'm out of town at the moment, but will be back by tomorrow night, so I'll add this on Wednesday morning.  Thank you for reviewing.

**Comment by yochannah** [comment]
@ChrisJefferson Thank you so much for volunteering to review! I'm hunting for a second reviewer (feel free to suggest if you know anyone), and once we find them we can kick off the full review. In the meantime feel free to look at the package and browse the review guidelines: https://joss.readthedocs.io/en/latest/review_criteria.html

**Comment by cicirello** [comment]
@ChrisJefferson I added a subsection to the repository's readme on building the library, as well as a directory (build) with an ant build file.  Assuming compilation is successful, the build will also execute all of the JUnit tests, and terminate if any tests fail (you'll see any relevant test results during the build, such as number of test cases, etc).  If you don't usually use ant, it can be obtained here: https://ant.apache.org/ 

For running the example programs, and the programs that replicate experimental data from published papers, I have updated the readme files in the examples and replication directories to include explicit instructions (including setting the classpath).

**Comment by motib** [comment]
I have agreed to review the submission although I haven't written Java for many years.....

I successfully installed ant and built the package. When I run the test case I get an error message:

D:\joss>java -classpath d:\joss\javaperm\lib;d:\joss\javaperm\testbin org.cicirello.permutations.PermutationTestCases
Error: Main method not found in class org.cicirello.permutations.PermutationTest
Cases, please define the main method as:
   public static void main(String[] args)
or a JavaFX application class must extend javafx.application.Application

@cicirello Could you please help?!

**Comment by yochannah** [comment]
Thanks @motib!

**Comment by yochannah** [comment]
@whedon assign @ChrisJefferson as reviewer

**Comment by whedon** [comment]
OK, the reviewer is @ChrisJefferson

**Comment by yochannah** [comment]
@whedon add @motib as reviewer

**Comment by whedon** [comment]
OK, @motib is now a reviewer
